code/datums/components/customizable_reagent_holder.dm

```diff
@@ -28,7 +28,7 @@
 		atom/replacement,
 		fill_type,
 		ingredient_type = CUSTOM_INGREDIENT_TYPE_EDIBLE,
-		max_ingredients = INFINITY,
+		max_ingredients = 98,
```

Comment:
Can you explain what this does to fix the bugs you've mentioned?

Reply:
overlay limit is 100

Reply:
Replace this with `MAX_ATOM_OVERLAYS - 2` then.